include/simdjson/dom/parser-inl.h

```diff
@@ -100,13 +100,18 @@ inline simdjson_result<element> parser::parse_into_document(document& provided_d
   // Important: It is possible that provided_doc is actually the internal 'doc' within the parser!!!
   error_code _error = ensure_capacity(provided_doc, len);
   if (_error) { return _error; }
-  std::unique_ptr<uint8_t[]> tmp_buf;
   if (realloc_if_needed) {
-    tmp_buf.reset(reinterpret_cast<uint8_t *>( internal::allocate_padded_buffer(len) ));
-    if (tmp_buf.get() == nullptr) { return MEMALLOC; }
-    std::memcpy(static_cast<void *>(tmp_buf.get()), buf, len);
+    // Make sure we have enough capacity to copy len bytes
+    if (_loaded_bytes_capacity < len) {
+      loaded_bytes.reset( internal::allocate_padded_buffer(len) );
+      if (!loaded_bytes) {
+        return MEMALLOC;
+      }
+      _loaded_bytes_capacity = len;
+    }
+    std::memcpy(static_cast<void *>(loaded_bytes.get()), buf, len);
```

Comment:
This seems to be unsafe in the sense that a pointer could be null.

Reply:
Yes, this could indeed happen if the input buffer was empty (len == 0). Fixed in new commit.
It could be that the parse file code path (not modified by this PR) has the same issue when a zero byte file is parsed, not sure.

Reply:
> It could be that the parse file code path (not modified by this PR) has a similar issue when a zero byte file is parsed, not sure. 

I will add a test for that specifically.

Reply:
See https://github.com/simdjson/simdjson/pull/1519